### ISSUE TITLE
Problem with uploading file (attachment)

### ISSUE THREAD

**Issue by OliDale** [stefan-niedermann/nextcloud-deck#646]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**

I can't add file in any way: from nextcloud, from phone, from google disk

**To reproduce**
Issue is reproducible with an account from [try.nextcloud.com](https://try.nextcloud.com):
- [*] Yes
- [ ] No

1. click + to add new task
2. fill title "test"
3. fill description "тест"
4. click "attachment"
5. add attachment from latest scrinshots
6. an error

another way, same problem
1. open gallery
2. choose photo (3 at a time in screenshot)
3. click "share"
4. click on "deck"
5. choose task

**Expected behavior**

Expected to add file

**Screenshots**

![Screenshot_20200805-130403_Nextcloud Deck](https://user-images.githubusercontent.com/59027495/89400743-fd7b6500-d71c-11ea-81c0-6f51add24aef.jpg)
![Screenshot_20200805-130449_Nextcloud Deck](https://user-images.githubusercontent.com/59027495/89400747-feac9200-d71c-11ea-9b24-4bc3f5407ba4.jpg)
![Screenshot_20200805-130631_Nextcloud Deck](https://user-images.githubusercontent.com/59027495/89400753-ff452880-d71c-11ea-830c-974b27cee5fc.jpg)
![Screenshot_20200805-130355_Nextcloud Deck](https://user-images.githubusercontent.com/59027495/89400755-ffddbf00-d71c-11ea-9f45-1906abb91e73.jpg)

**Versions**
 - Nextcloud:   NextcloudPi 18.0.7
 - Nextcloud Deck: 1.0.5
 - Nextcloud Android: 3.12.1
 - Nextcloud Android Deck: 1.9.6

**Smartphone (please complete the following information):**
- Device:  Samsung Galaxy S9+
- Android-Version: 10
- App-Store:
  - [*] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [ ] F-Droid
  - [ ] Huawei AppGallery

**Are you using LDAP?**
- [ ] Yes
- [ ] No
- [*] i dont know

**Stacktrace**

```
App Version: 1.9.6
App Version Code: 1009006
App Flavor: play

Deck Server Version: 1.0.5

Files App Version Code: 30120190

---

OS Version: 4.9.118-18847185(G965FXXUAETG3)
OS API Level: 29
Device: star2lte
Manufacturer: samsung
Model (and Product): SM-G965F (star2ltexx)

---

it.niedermann.nextcloud.deck.exceptions.UploadAttachmentFailedException: Unknown URI scheme
at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.lambda$onActivityResult$1$CardAttachmentsFragment(CardAttachmentsFragment.java:226)
at it.niedermann.nextcloud.deck.ui.card.attachments.-$$Lambda$CardAttachmentsFragment$gVs3JFN_5PB5-DoJ7FGcuWVctUg.onChanged(Unknown Source:8)
at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.util.LiveDataHelper$2.onChanged(LiveDataHelper.java:79)
at androidx.lifecycle.LiveData.considerNotify(LiveData.java:131)
at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:149)
at androidx.lifecycle.LiveData.setValue(LiveData.java:307)
at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
at androidx.lifecycle.LiveData$1.run(LiveData.java:91)
at android.os.Handler.handleCallback(Handler.java:883)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8125)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: com.nextcloud.android.sso.exceptions.NextcloudHttpRequestFailedException: Ошибка выполнения запроса HTTP. Код состояния: 500
at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:172)
at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:153)
at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:97)
at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$HPzOdi9LOc_f-wA9GNnhc9LKQSM.subscribe(Unknown Source:6)
at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
at io.reactivex.Observable.subscribe(Observable.java:12284)
at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
at io.reactivex.Observable.subscribe(Observable.java:12284)
at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.IllegalStateException: {"status":500,"message":"\u0424\u0430\u0439\u043b \u043d\u0435 \u043f\u043e\u043b\u0443\u0447\u0435\u043d \u0438\u043b\u0438 \u0435\u0433\u043e \u0440\u0430\u0437\u043c\u0435\u0440 \u043f\u0440\u0435\u0432\u044b\u0448\u0430\u0435\u0442 150 GB"}

at com.nextcloud.android.sso.InputStreamBinder.processRequestV2(InputStreamBinder.java:429)
at com.nextcloud.android.sso.InputStreamBinder.performNextcloudRequestAndBodyStreamV2(InputStreamBinder.java:125)
at com.nextcloud.android.sso.InputStreamBinder.performNextcloudRequestV2(InputStreamBinder.java:108)
at com.nextcloud.android.sso.aidl.IInputStreamService$Stub.onTransact(IInputStreamService.java:134)
at android.os.Binder.execTransactInternal(Binder.java:1056)
at android.os.Binder.execTransact(Binder.java:1029)

```

**Comment by stefan-niedermann** [comment]
Thank you! I will have a look as soon as possible, might be an android permission issue. Is this only since the latest version?

**Comment by desperateCoder** [comment]
Hi @OliDale, thanks for your report!
Sorry for the inconveniences, we would like to resolve this. 

You posted this screenshot:

> [stacktrace](https://user-images.githubusercontent.com/59027495/89400747-feac9200-d71c-11ea-9b24-4bc3f5407ba4.jpg)

Can you please copy the text and post it here completely? The relevant part for us isn't visible in the screenshot.

**Comment by stefan-niedermann** [comment]
He sent me the full stacktrace via mail, i will add it here later

**Comment by stefan-niedermann** [comment]
```
App Version: 1.9.6
App Version Code: 1009006
App Flavor: play

Deck Server Version: 1.0.5

Files App Version Code: 30120190

---

OS Version: 4.9.118-18847185(G965FXXUAETG3)
OS API Level: 29
Device: star2lte
Manufacturer: samsung
Model (and Product): SM-G965F (star2ltexx)

---

it.niedermann.nextcloud.deck.exceptions.UploadAttachmentFailedException: Unknown URI scheme
at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.lambda$onActivityResult$1$CardAttachmentsFragment(CardAttachmentsFragment.java:226)
at it.niedermann.nextcloud.deck.ui.card.attachments.-$$Lambda$CardAttachmentsFragment$gVs3JFN_5PB5-DoJ7FGcuWVctUg.onChanged(Unknown Source:8)
at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.util.LiveDataHelper$2.onChanged(LiveDataHelper.java:79)
at androidx.lifecycle.LiveData.considerNotify(LiveData.java:131)
at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:149)
at androidx.lifecycle.LiveData.setValue(LiveData.java:307)
at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
at androidx.lifecycle.LiveData$1.run(LiveData.java:91)
at android.os.Handler.handleCallback(Handler.java:883)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8125)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: com.nextcloud.android.sso.exceptions.NextcloudHttpRequestFailedException: Ошибка выполнения запроса HTTP. Код состояния: 500
at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:172)
at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:153)
at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:97)
at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$HPzOdi9LOc_f-wA9GNnhc9LKQSM.subscribe(Unknown Source:6)
at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
at io.reactivex.Observable.subscribe(Observable.java:12284)
at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
at io.reactivex.Observable.subscribe(Observable.java:12284)
at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.IllegalStateException: {"status":500,"message":"\u0424\u0430\u0439\u043b \u043d\u0435 \u043f\u043e\u043b\u0443\u0447\u0435\u043d \u0438\u043b\u0438 \u0435\u0433\u043e \u0440\u0430\u0437\u043c\u0435\u0440 \u043f\u0440\u0435\u0432\u044b\u0448\u0430\u0435\u0442 150 GB"}

at com.nextcloud.android.sso.InputStreamBinder.processRequestV2(InputStreamBinder.java:429)
at com.nextcloud.android.sso.InputStreamBinder.performNextcloudRequestAndBodyStreamV2(InputStreamBinder.java:125)
at com.nextcloud.android.sso.InputStreamBinder.performNextcloudRequestV2(InputStreamBinder.java:108)
at com.nextcloud.android.sso.aidl.IInputStreamService$Stub.onTransact(IInputStreamService.java:134)
at android.os.Binder.execTransactInternal(Binder.java:1056)
at android.os.Binder.execTransact(Binder.java:1029)

```

**Comment by OliDale** [comment]
> Thank you! I will have a look as soon as possible, might be an android permission issue. Is this only since the latest version?

Only in latest version. Because I installed it only today.
Also I have crash report via try.nextcloud.com:
App Version: 1.9.6
App Version Code: 1009006
App Flavor: play

Deck Server Version: 1.0.5

Files App Version Code: 30120190

---

OS Version: 4.9.118-18847185(G965FXXUAETG3)
OS API Level: 29
Device: star2lte
Manufacturer: samsung
Model (and Product): SM-G965F (star2ltexx)

---

it.niedermann.nextcloud.deck.exceptions.UploadAttachmentFailedException: Unknown URI scheme
at it.niedermann.nextcloud.deck.ui.card.attachments.CardAttachmentsFragment.lambda$onActivityResult$1$CardAttachmentsFragment(CardAttachmentsFragment.java:226)
at it.niedermann.nextcloud.deck.ui.card.attachments.-$$Lambda$CardAttachmentsFragment$gVs3JFN_5PB5-DoJ7FGcuWVctUg.onChanged(Unknown Source:8)
at it.niedermann.nextcloud.deck.persistence.sync.adapters.db.util.LiveDataHelper$2.onChanged(LiveDataHelper.java:79)
at androidx.lifecycle.LiveData.considerNotify(LiveData.java:131)
at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:149)
at androidx.lifecycle.LiveData.setValue(LiveData.java:307)
at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
at androidx.lifecycle.LiveData$1.run(LiveData.java:91)
at android.os.Handler.handleCallback(Handler.java:883)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loop(Looper.java:237)
at android.app.ActivityThread.main(ActivityThread.java:8125)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: com.nextcloud.android.sso.exceptions.NextcloudHttpRequestFailedException: Ошибка выполнения запроса HTTP. Код состояния: 500
at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:172)
at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:153)
at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:97)
at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$HPzOdi9LOc_f-wA9GNnhc9LKQSM.subscribe(Unknown Source:6)
at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
at io.reactivex.Observable.subscribe(Observable.java:12284)
at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
at io.reactivex.Observable.subscribe(Observable.java:12284)
at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.IllegalStateException: {"status":500,"message":"\u0424\u0430\u0439\u043b \u043d\u0435 \u043f\u043e\u043b\u0443\u0447\u0435\u043d \u0438\u043b\u0438 \u0435\u0433\u043e \u0440\u0430\u0437\u043c\u0435\u0440 \u043f\u0440\u0435\u0432\u044b\u0448\u0430\u0435\u0442 2 MB"}

at com.nextcloud.android.sso.InputStreamBinder.processRequestV2(InputStreamBinder.java:429)
at com.nextcloud.android.sso.InputStreamBinder.performNextcloudRequestAndBodyStreamV2(InputStreamBinder.java:125)
at com.nextcloud.android.sso.InputStreamBinder.performNextcloudRequestV2(InputStreamBinder.java:108)
at com.nextcloud.android.sso.aidl.IInputStreamService$Stub.onTransact(IInputStreamService.java:134)
at android.os.Binder.execTransactInternal(Binder.java:1056)
at android.os.Binder.execTransact(Binder.java:1029)

**Comment by OliDale** [comment]
Here is scrinshot from my phone:
As you can see it has no permissions to files and camera (did't ask on first start)
![Screenshot_20200805-140433_Permission controller](https://user-images.githubusercontent.com/59027495/89405942-0cfeac00-d725-11ea-9c92-422ea8443d19.jpg)

**Comment by stefan-niedermann** [comment]
Thank you for your help, this is the key. Will be fixed in the next version.

**Comment by desperateCoder** [comment]
@stefan-niedermann SSO fix is implemented, you can merge the branch and release (Jitpack dependency!)

**Comment by stefan-niedermann** [comment]
Fix shipped in version `1.9.7`.
